clesperanto/plugins/add_images_weighted.py

```diff
@@ -1,7 +1,11 @@
 
 from ..core import execute
+from ..core import create_like
+from ..core import plugin_function
+from gputools import OCLArray
 
-def add_images_weighted(input1, input2, output, weight1, weight2):
+@plugin_function(output_creator=create_like)
+def add_images_weighted(input1:OCLArray, input2:OCLArray, output:OCLArray, weight1=1, weight2=1):
```

Comment:
Here I would make the types:

```python
Array = Union[np.ndarray, OCLArray]

@plugin_function(output_creator=create_like)
def add_images_weighted(
    image1 : Array,
    image2 : Array,
    *,  # <- this means that later arguments are keyword-only, ie you have to write out=mybuffer
    out : Optional[Array] = None,
    weight1 : float = 1,
    weight2 : float = 1,
):
```

Reply:
Hey @jni,

"keyword only" means that 
```
add_images_weighted(in1, int2, out, 1, 1)
```
wouldn't work anymore?

Reply:
Yes.

Reply:
Hey @lagru 👋 

pitty. That would mean that [this](https://clesperanto.github.io/images/clijmapy.mp4) wouldn't work - the major reason of the whole exercise. ;-)

So I guess we have to live without that feature. At least on this abstraction layer....

Reply:
Which feature are you talking about exactly? Are you saying you are lazy and don't want to tell me your parameter names to make sure that they are clear to read? ;)

Reply:
With "feature" I mean specifying parameter names is optional. 

This needs to work (Reason: Make the code run in other languages. clEsperanto)
```
add_images_weighted(in1, int2, out, 1, 1)
```

and this is optional:
```
add_images_weighted(in1, int2, out, weight1=1, weight2=1)
```

Is there any secret pythonic rule I violate with that?

Reply:
It's just good practice in "modern" Python to say that the names should always be there. It's easy enough to strip the names out in other languages? Other languages will probably use the name `addImagesWeighted` anyway and so on, no?

Reply:
Java-esque languages (Java, Matlab, JavaScript) don't like the `method(parameter=value)` pattern afaik. To circumvent that, developers prefer the builder pattern `method().setParameter(value).run()`. Which is nice but also complicated, especially for coding newbies. 
 
I'd like to keep things simple - and copy-pastable between languages. Thus, all  users from all languages have to step a little step back and use very simple method calls.

Btw. the JavaScript recorder records something like
```
image1 = ...;
image2 = ....;
output = ....;
weight1 = 1;
weight2 = 1;
add_images_weighted(image1, image2, output, weight1, weight2);
```
And this should run ins all target languages. 

I suggest to leave it as it is for now. But any idea/suggestion is welcome!

Reply:
I honestly think that copy-pasting code between languages should not be a goal, only making the code recognisable between languages. When in Rome, do as the Romans do, and idiomatic Python is different from idiomatic Java. Completely unifying their syntax is, in my opinion, a recipe for a compromise where everyone is unhappy. Not only that, but people seeing code like this when they are learning a language will then learn antipatterns for that language.

As mentioned elsewhere, we can postpone this decision, that's fine.